### ISSUE TITLE
Lesson 007 - Advanced Layout & Your Résumé

### ISSUE THREAD

**Issue by devinleonhart** [devinleonhart/learn-web-development#7]
# Lesson 007 - Advanced Layout & Résumé

This is the first lesson where we are going to focus on a product. We're going to make something relatively simple, something that shows you off. People who create things need a portfolio to prove they can do what they do to future employers. The empty promise of being open to new ideas and working hard means very little.

In an interview, what sells you in the moment, is your current surface knowledge of the industry and your past accomplishments.

This creates a terrible catch-22 with experience, but that's in a lot of industries. What's actually worse is the surface knowledge part. There is a ton of stuff in our industry and it is very easy to find details that an experienced developer has never heard before. The way we develop resiliency to this kind of treatment and get our dream jobs is by making a larger and larger portfolio. 

If someone says "How does asynchronous code in Javascript work?" in an interview, if you did not recently work on this, you're likely going to miss the details the interviewer really cares about. A portfolio allows you to change the answer into something concrete. "Well, I worked with that on X. I have it right here. Let's go through it and see if it's something you're looking for." 

You become better armed to deal with questions about your skill when your accomplishments are following you. And if the employer seems hellbent about making you answer the questions in a vacuum, like you should know 100% of a field in surface memory, it's likely just a bullshit interview and you should move on.

## Résumé Project
We're going to make the first project into one that doesn't have a lot of interactivity. It's something meant to just show you off. You link it as if giving away a business card. We're going to call the project "bio" for short.

This resume page will have all of the key contact information about you and a list of your pervious work experience. However what makes this special, apart from say, a word document, is that we can provide examples of your accomplishments in links. 

Say you wanted to prove you understand asynchronous Javascript or CSS animations, instead of just saying you do on a list, we're going to have sections of this resume app that use them and demonstrate competency. If they ask "How did you build that?" you open up this repo and show them. Or... you know, just answer from surface memory if you remember it all.

## Advanced Layout
You have learned how to make a layout in CSS with the concept of `block` and `inline`. But if that's all you had to work with, you'd have a lot of fiddly manual changes to do. Right now, one of the best things to learn for advanced Layout techniques is the use of [Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)

Knowing Flexbox actually got me a job once. Knowing how it worked from surface memory, thanks to a recent project I'd worked on, really helped. You'll notice that it has its own display type `display: flex;`

The idea of Flexbox is to have two different pieces of the page working together in order to make the layout respond to screen size change and redraw. The first part is the `flex container`, which is set to `display: flex;` and the second part are all of the things inside the container, called the `flex items`. 

Dropping Flexbox onto someone new is a little cruel. But I think you can handle it. Also, knowledge of this step is not a warmup. This is a marketable skill. If you work with this part and enjoy it, you can get paid to keep doing it. Please keep that in mind.

## 2-Column Layouts
Now, we're getting used to arranging boxes. But what if we didn't give the constraint of 100px for length and width? What if we said they could be as big as the screen? When a box is big enough, it's not treated like a single item anymore. They are treated like big "sections" of the screen. If you have one box that's as tall as the screen yet only half the width, that's basically a column. It divides the screen in half. Anything in that box is in one half and everything not in that box is in the other. 

The homework will ask you to make a two column layout. The way to solve this is to think of each column as a "box" that is big enough that people don't realize it's a box anymore. You will need Flexbox to pull this off. A giant source of help will be the CSS Tricks example where they use `aside` HTML elements. Don't be intimidated, though. We'll cover what that is later.

## Assignment

- [ ] Make two folders in your `learn-web-development` repo. Name one "bio" and the other "scratchpad". You may change these names to whatever you want.
- [ ] Move the three files `index.html`, `index.js`, `main.css` to the scratchpad. The scratchpad area will now be for trying things and experimenting.
- [ ] Make three new files, with the same names, and copy them to the bio folder. You should now have no files in the root. Just two folders. Bio will be for our finished product.
- [ ] Add the required HTML elements to the bio `index.html`. This is the `html` element along with the `head`, `title` and `body` elements in the right place. Link the other two files with `link` and `script`. A good blank canvas to start with. Once you're sure it's right, make a single commit with all of the changes above. This is now the "starting point" of the new project.
- [ ] Use Flexbox to make a horizontal row of boxes on the scratchpad. The examples on the CSS Tricks site should be able to help you. Remember that since it's a scratchpad, you can add or remove anything else you want.
- [ ] After you feel you have gotten Flexbox down, make a two column layout with a header and footer. Remember to check CSS tricks for the example of how to do this. Commit this result. It can help to do what CSS tricks did in their example, color coding each piece with `background-color` to make sure you know where things are. You also can optionally use a `CSS reset` from the previous lesson.

Good luck.

**Comment by devinleonhart** [comment]
@daveleonhart Assignment is up.

**Comment by daveleonhart** [comment]
@devinleonhart 

e8a5f42b3cdcb142b246fd752898fef145c490b3


This may very well be a "simulated correct answer". I'm still not fully confident with Flexbox yet, but I'm super proud of getting this far.

**Comment by devinleonhart** [comment]
Nope. That's correct. The bio uses two columns, 50%, and has a header and a footer that are not affected by the use of flexbox. That's what I was looking for. 

The scratchpad one is difficult to read because, despite them being divs, they are only as large as a single character. This disguises the "boxes" and makes them all blur into a single blue line. It could be better illustrated by making the boxes bigger and only coloring the internals, instead of the line behind them.

Good work. That was not an easy concept to grasp with so little buildup.
